client-tez/src/main/java/org/apache/tez/common/TezClientConf.java

```diff
@@ -0,0 +1,395 @@
+          .booleanType()
+          .defaultValue(false)
+          .withDescription("");
+
+  public static final ConfigOption<Integer> RSS_CLIENT_ASSIGNMENT_SHUFFLE_SERVER_NUMBER =
+      ConfigOptions.key("rss.client.assignment.shuffle.nodes.max")
+          .intType()
+          .defaultValue(-1)
+          .withDescription("");
+
+  public static final ConfigOption<Integer> RSS_SHUFFLE_SOURCE_VERTEX_ID =
+      ConfigOptions.key("tez.rss.shuffle.source.vertex.id")
+          .intType()
+          .defaultValue(-1)
+          .withDescription("");
+
+  public static final ConfigOption<Integer> RSS_SHUFFLE_DESTINATION_VERTEX_ID =
+      ConfigOptions.key("tez.rss.shuffle.destination.vertex.id")
+          .intType()
+          .defaultValue(-1)
+          .withDescription("");
+
+  public static final ConfigOption<String> RSS_SHUFFLE_MODE =
+      ConfigOptions.key("tez.shuffle.mode").stringType().defaultValue("remote").withDescription("");
+
+  public static final ConfigOption<String> RSS_REMOTE_SPILL_STORAGE_PATH =
+      ConfigOptions.key("tez.rss.remote.spill.storage.path")
+          .stringType()
+          .defaultValue("")
+          .withDescription("");
+
+  public static final ConfigOption<Boolean> RSS_REDUCE_REMOTE_SPILL_ENABLED =
+      ConfigOptions.key("tez.rss.reduce.remote.spill.enable")
+          .booleanType()
+          .defaultValue(false)
+          .withDescription("");
+
+  public static final ConfigOption<Integer> RSS_REDUCE_REMOTE_SPILL_REPLICATION =
+      ConfigOptions.key("tez.rss.reduce.remote.spill.replication")
+          .intType()
+          .defaultValue(1)
+          .withDescription("");
+
+  public static final ConfigOption<Integer> RSS_REDUCE_REMOTE_SPILL_RETRIES =
+      ConfigOptions.key("tez.rss.reduce.remote.spill.retries")
+          .intType()
+          .defaultValue(5)
+          .withDescription("");
+
+  public static final ConfigOption<Boolean> TEZ_RSS_TEST_MODE_ENABLE =
+      ConfigOptions.key("tez.rss.test.mode.enable")
+          .booleanType()
+          .defaultValue(false)
+          .withDescription("Whether enable test mode for the shuffle server.");
+
+  public static final Set<String> RSS_MANDATORY_CLUSTER_CONF =
+      ImmutableSet.of(RSS_STORAGE_TYPE.key(), RSS_REMOTE_STORAGE_PATH.key());
+
+  public Configuration getHadoopConfig() {
+    return hadoopConfig;
+  }
+
+  private final Configuration hadoopConfig;
+
+  public TezClientConf() {
+    this(new Configuration());
+  }
+
+  public TezClientConf(Configuration config) {
+    super();
+    boolean ret = loadConfFromHadoopConfig(config);
+    if (!ret) {
+      throw new IllegalStateException("Fail to load config " + config);
+    }
+    this.hadoopConfig = config;
+  }
+
+  public boolean loadConfFromHadoopConfig(Configuration config) {
+    return loadConfFromHadoopConfig(config, ConfigUtils.getAllConfigOptions(TezClientConf.class));
+  }
+
+  public RssConf toRssConf() {
```

Comment:
TezClientConf is extended from RssBaseConf, RssBaseConf is extended from RssConf. So is this method still necessary?

Reply:
I see you! 
But I don't know the design of these issue. How about RssClientConfig? 
Maybe RssClientConfig should be extended from RssConf or RssBaseConf. RssTezConfig should be extended from RssClientConfig.
The the config key in RssXXXConfig should not be start with "tez", and the config key in hadoop Configuration should be start with "tez". That's my understanding, maybe it is wrong.

What do you think? @lifeSo 

@zuston Can you give us some advice?

Reply:
In now design, the config in Tez client, start with "tez.", 
similar, the config in Spark client, start with "spark.".
And there is also leading substring in RssSparkConfig:



```
  public static RssConf toRssConf(SparkConf sparkConf) {
    RssConf rssConf = new RssConf();
    for (Tuple2<String, String> tuple : sparkConf.getAll()) {
      String key = tuple._1;
      if (!key.startsWith(SPARK_RSS_CONFIG_PREFIX)) {
        continue;
      }
      key = key.substring(SPARK_RSS_CONFIG_PREFIX.length());
      rssConf.setString(key, tuple._2);
    }
    return rssConf;
  }
```

Reply:
If some tez config not start with "tez.", it will be confusing because some default tez config is start with 'tez.'.

eg:

https://github.com/apache/tez/blob/master/tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java